torch/csrc/jit/python_tracer.h

```diff
@@ -29,7 +29,7 @@ std::shared_ptr<Graph> createGraphByTracing(
     TypedStack inputs,
     const py::function& var_name_lookup_fn,
     bool force_outplace,
-    const std::shared_ptr<script::Module>& self = nullptr);
+    script::Module* self = nullptr);
```

Comment:
There are a couple places in this PR where we take `Module*`, I assume so that it can be null. I  think it's a bit weird pass around a pointer to a view of a reference type…maybe `c10::optional`?

Reply:
The places where Module* are used are trying to avoid dependencies in headers. For instance, the tracer.h header is included inside of VariableType.h, a huge file. We are careful to avoid including Module.h because then it would rebuild a huge part of pytorch when modified. Unfortunately using c10::optional requires the definition to be present, so Module* is a workaround that does the same thing.